presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergAbstractMetadata.java

```diff
@@ -742,10 +755,65 @@ public void truncateTable(ConnectorSession session, ConnectorTableHandle tableHa
     public ConnectorTableHandle beginDelete(ConnectorSession session, ConnectorTableHandle tableHandle)
     {
         IcebergTableHandle handle = (IcebergTableHandle) tableHandle;
+
         if (handle.isSnapshotSpecified()) {
             throw new PrestoException(NOT_SUPPORTED, "This connector do not allow delete data at specified snapshot");
         }
-        throw new PrestoException(NOT_SUPPORTED, "This connector only supports delete where one or more partitions are deleted entirely");
+        if (getDeleteMode(getTableMetadata(session, handle).getProperties()) == RowLevelOperationMode.COPY_ON_WRITE) {
+            throw new PrestoException(NOT_SUPPORTED, "This connector only supports delete where one or more partitions are deleted entirely");
+        }
+
+        Table icebergTable = getIcebergTable(session, handle.getSchemaTableName());
+        validateTableMode(session, icebergTable);
+        transaction = icebergTable.newTransaction();
+
+        return handle;
+    }
+
+    @Override
+    public void finishDelete(ConnectorSession session, ConnectorTableHandle tableHandle, Collection<Slice> fragments)
+    {
+        IcebergTableHandle handle = (IcebergTableHandle) tableHandle;
+        Table icebergTable = getIcebergTable(session, handle.getSchemaTableName());
+
+        RowDelta rowDelta = transaction.newRowDelta();
+
+        List<CommitTaskData> commitTasks = fragments.stream()
+                .map(slice -> commitTaskCodec.fromJson(slice.getBytes()))
+                .collect(toImmutableList());
+
+        Type[] partitionColumnTypes = icebergTable.spec().fields().stream()
+                .map(field -> field.transform().getResultType(
+                        icebergTable.schema().findType(field.sourceId())))
+                .toArray(Type[]::new);
+
+        CharSequenceSet referencedDataFiles = CharSequenceSet.empty();
+
+        for (CommitTaskData task : commitTasks) {
+            FileMetadata.Builder builder = FileMetadata.deleteFileBuilder(icebergTable.spec())
```

Comment:
The commitTask may use a historical spec version, so could we simply use `icebergTable.spec()` here?

Reply:
@hantangwangd, thank you for pointing these out! 

I have handled your review comments. Please take a look when you get a chance, thanks!